Add a Pull Request template, and pull request documentation

xfrenette-crim opened the pull request:
## Description

When working with multiple developers, the _Pull Request process_ (reviewing, approving, merging) should be documented to ensure the same process is followed by everyone to prevent any miscommunication and ensure a more efficient process.

This PR adds documentation about the Pull Request process in CONTRIBUTING.md. It also introduces a simple _GitHub Pull Request template_ for the future PRs.

Note that GitHub requires that the Pull Request template be in the `main` branch, which is why this PR is to be merged in `main`.

MahieddineAk commented on CONTRIBUTING.md:
```diff
@@ -23,3 +23,44 @@
+1. Update the code as requested, if changes were requested.
+2. Mark as "resolved" all pending discussions.
+3. **_Squash and Merge_** the PR.
+4. Delete the branch, if it's a _feature_ or _fix_ branch.
```
@xfrenette-crim reading that point suggests there is a case where the branch doesn't have to be deleted (like `demo-2.0`), you may want to clarify that point.

MahieddineAk commented on CONTRIBUTING.md:
```diff
@@ -23,3 +23,44 @@
+**Notes**:
+* Make sure to read the comments left by the reviewer. Even if the PR is approved, there might be requests for minor changes.
+* Beside the requested minor changes, don't update your code after the PR has been approved.
+* Always _Squash and Merge_, unless exception.
```
@xfrenette-crim you may want to clarify which are the exceptions.

MahieddineAk commented on CONTRIBUTING.md:
```diff
@@ -23,3 +23,44 @@
+* Make sure to read the comments left by the reviewer. Even if the PR is approved, there might be requests for minor changes.
+* Beside the requested minor changes, don't update your code after the PR has been approved.
+* Always _Squash and Merge_, unless exception.
+* If a requested change needs more discussion, discuss under the comment left by the reviewer. If a discussion happened outside of GitHub
```
@xfrenette-crim for some discussion like on Slack, we can attach the thread link.

MahieddineAk commented on CONTRIBUTING.md:
```diff
@@ -23,3 +23,44 @@
+
+The person that opened the PR will then be responsible to:
+1. Update the code as requested (but don't "resolve" the reviewer's comments).
+2. Request again a review from the reviewer (using the button beside the reviewer's name).
```
@xfrenette-crim I think you mean the button with arrows, right?

MahieddineAk commented on CONTRIBUTING.md:
```diff
@@ -23,3 +23,44 @@
+  nor simply send a message to the reviewer.
+* **_Don't_** mark the reviewer's comments as "Resolved" after you updated your code. The reviewer will need to know
+  what was the comment before re-validating it. The reviewer will "resolve" the conversation once it's considered resolved.
+  If you want to show that you have now "resolved" the comment, you can add a reaction (ex: 👍) or a comment to the comment.
```
@xfrenette-crim I think sying "comment to the thread" will be more clear.

MahieddineAk reviewed (commented):
@xfrenette-crim I've left you minor comments. Also, you may want to specify that the language to be used is English, so all developers will be able to engage with and understand any PR, especially since not all of them know French.

perronld commented:
Looks good to me. Except that I would add a very telegraphic version of the numbered steps pull_request_template.md as a reminder, especially about when to resolve and not to resolve conversations. For people who works on multiple projects, it can be easy to skip those.

xfrenette-crim commented:
@MahieddineAk I updated the documentation:

* I added a little bit more details in CONTRIBUTING.md to make some things clearer (based on your comments).
* I also added a new section for the "PR reviewer", and adjusted the section for the "PR creator".
* As suggested by Louis-David, I added a summary of the PR process in the Pull Request template. Since the template is first seen as plain text by the PR creator, I removed a little bit of Markdown formatting, to make it easier to read in plain text.
* I also ensured maximum line widths. For the template, I limited it to 80 characters to better fit in the text field of the PR description.

MahieddineAk reviewed (approved):
@xfrenette-crim this is looking good to me, thanks!
